File: leaf-system/src/main/java/com/leaf/system/service/LeafRoleService.java
package com.leaf.system.service;

import com.leaf.framework.record.PageRecord;
import com.leaf.system.domain.LeafRole;
import com.leaf.system.domain.LeafRoleTable;
import jakarta.enterprise.context.ApplicationScoped;
import org.babyfish.jimmer.Page;
import org.babyfish.jimmer.sql.JSqlClient;

/**
 * @author jhlz
 * @version 1.0.0
 */
@ApplicationScoped
public class LeafRoleService {
    private final JSqlClient sqlClient;
    LeafRoleTable table = LeafRoleTable.$;

    public LeafRoleService( JSqlClient sqlClient) {
        this.sqlClient = sqlClient;
    }

    public LeafRole save(LeafRole leafRole) {
        return sqlClient.save(leafRole).getModifiedEntity();
    }

    public LeafRole findById(long id) {
        return sqlClient.findById(LeafRole.class, id);
    }

    public Page<LeafRole> page(PageRecord page) {
        System.out.println(page);
        return null;
        // return sqlClient.createQuery(table)
        //         .select(table)
        //         .fetchPage(page.page() - 1, page.size());
    }
}

File: src/main/kotlin/com/jhlz/service/LeafRoleService.kt
package com.jhlz.service

import com.jhlz.domain.LeafRole
import com.jhlz.domain.fetchBy
import com.jhlz.record.PageQuery
import jakarta.inject.Singleton
import org.babyfish.jimmer.Page
import org.babyfish.jimmer.sql.kt.KSqlClient

/**
 * @author jhlz
 * @version x.x.x
 */
@Singleton
class LeafRoleService (val sqlClient: KSqlClient){
    fun page(pageQuery: PageQuery): Page<LeafRole> {
        return sqlClient.createQuery(LeafRole::class) {
            select(
                table.fetchBy {
                    allScalarFields()
                    creator { username() }
                }
            )
        }.fetchPage(pageQuery.pageNum - 1, pageQuery.pageSize)
    }

    fun info(id: Long): LeafRole {
        return sqlClient.findById(LeafRole::class, id)!!
    }

    fun add(role: LeafRole) {
        sqlClient.entities.save(role)
        throw Exception("test")

    }

    fun update(role: LeafRole) {
        sqlClient.entities.save(role)
    }

    fun delete(ids: List<Long>) {
        sqlClient.deleteByIds(LeafRole::class, ids)
    }
}